symfit/core/fit.py

```diff
@@ -1626,4 +1623,81 @@ def hessian_from_model(model):
-    return jacobian_from_model(jac_model)
\ No newline at end of file
+    return jacobian_from_model(jac_model)
+
+
+def prepare_constraint(relational, extra_params, model_type=CallableModel):
+    """
+    Based on a relational expression, return a Model representing the
+    constraint.
+
+    :param relational: Expression which is a subclass of :class:`~sympy.Rel`.
+    :param extra_params: Parameters for the constraint. Usually this should be
+        the parameters of the corresponding model.
+    :param model_type: Type of model for the constraint.
+    :return: Model representing the constraint.
+    """
+    allowed_types = [sympy.Eq, sympy.Ge, sympy.Le]
+    constraint_type = type(relational)
+    # Check if the type of each constraint is allowed.
+    if constraint_type not in allowed_types:
+        raise ModelError(
+            'Only constraints of the type {} are allowed. A constraint'
+            ' of type {} was provided.'.format(allowed_types,
+                                               constraint_type)
+        )
+    elif constraint_type is sympy.Le:
+        # We change this to a Ge
+        constraint = relational.rhs - relational.lhs
+        constraint_type = sympy.Ge
+    else:
+        constraint = relational.lhs - relational.rhs
+
+    if not issubclass(model_type, BaseNumericalModel):
+        constraint_model = model_type(constraint)
+    else:
+        y = Variable()
+        vars, params = seperate_symbols(constraint)
+        connectivity_mapping = {y: set(vars + params)}
+        constraint_model = model_type({y: constraint},
+                                      connectivity_mapping=connectivity_mapping)
+
+    constraint_model.params = extra_params
+    constraint_model.constraint_type = constraint_type
+    return constraint_model
+
+def constraints_from_relations(relationals, model):
+    """
+    Turn Relationals into models which can serve as constraints.
+
+    :param relationals:
+    :param model:
+    :return:
+    """
+    constraints = []
+    for relational in relationals:
+        constraint_model = prepare_constraint(relational, model.params,
+                                              model_type=model.__class__)
+        if any(var in model for var in constraint_model.independent_vars):
+            # This constraint depends on the output of the model, so we need to
+            # work the model into the constraint
+            model_dict = constraint_model.model_dict.copy()
+            constraint_type = constraint_model.constraint_type
+            for var in constraint_model.independent_vars:
+                if var in model:
+                    model_dict[var] = model[var]
+                    # Also add possible dependencies of this var
+                    for symbol in model.ordered_symbols:
+                        # Walk over all possible dependencies of this
+                        # variable until we no longer have dependencies.
```

Comment:
Does this also deal with dependencies of dependencies of dependencies ...?

Reply:
Yes, due to the ordering everything is passed on because we start with the low hanging fruit and work our way up